File: src/compiler/lexer/token/WhitespaceToken.java
package compiler.lexer.token;

import compiler.parser.TokenEvaluator;
import compiler.parser.visitors.TokenVisitor;

public final class WhitespaceToken extends Token {
  public WhitespaceToken(String str) {
    super(str, 100);
  }

  @Override
  public <T> T accept(TokenVisitor<T> visitor) {
    return visitor.visit(this);
  }

  @Override
  public Void accept(TokenEvaluator visitor) throws Exception {
    return visitor.visit(this);
  }
}

File: src/compiler/lexer/token/WhitespaceToken.kt
package compiler.lexer.token

import compiler.parser.TokenEvaluator
import compiler.parser.visitors.TokenVisitor

class WhitespaceToken(str: String) : Token(str, 100) {
    override fun <T> accept(visitor: TokenVisitor<T>): T {
        return visitor.visit(this)
    }

    @Throws(Exception::class)
    override fun accept(visitor: TokenEvaluator) {
        return visitor.visit(this)
    }
}
